### ISSUE TITLE
Add IAU standard designation from coordinate mathod?

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#8861]
When making source catalogs, one usually has to compute source designation strings from the measured source sky coordinates. Doing this "right", i.e. following http://cdsweb.u-strasbg.fr/Dic/iau-spec.html, is a bit tricky when it comes to rounding.

Years ago I wrote some code to do this and added it in Gammapy, but now that we clean up to prepare for Gammapy v1.0, we remove such code that is not gamma-ray specific or appropriate for Gammapy.

I'm not sure if it's easy to do this task by now with `astropy.coordinates`. If yes, I'd suggest adding a short docs section how to do it. If no, maybe it would be in scope to add a helper function or method to `SkyCoord` to format to this string?

See Gammapy: [coordinate_iau_format](https://docs.gammapy.org/0.12/api/gammapy.catalog.coordinate_iau_format.html), [code](https://github.com/gammapy/gammapy/blob/697fa5c4c5153a9f6fb8e7360b71eab58ac6ab0d/gammapy/catalog/utils.py#L16-L231), [tests](https://github.com/gammapy/gammapy/blob/697fa5c4c5153a9f6fb8e7360b71eab58ac6ab0d/gammapy/catalog/tests/test_utils.py#L1-L83)

I'd be happy to send a PR for this to Astropy, adding a method or a short docs section, if you think it's useful to add.

**Comment by pllim** [comment]
cc @astropy/coordinators

**Comment by mhvk** [comment]
Yes, please! Would it be a method/attribute, or just a special `style` in `to_string`? Arguably, we should have the latter in any case, perhaps with a more easily discoverable short-cut. 

This brings up a related question, whether, if we initialize a `SkyCoord` from such a string, should account for the truncate-rather-than-round nature of those. My sense is that we should, but it might add to confusion...
